packages/frontend/src/components/Explorer/ExportGsheets/index.tsx

```diff
@@ -26,49 +30,67 @@ const ExportGsheets: FC<ExportGsheetProps> = memo(
 
         const { showToastError, showToast } = useToaster();
 
-        const { mutateAsync: exportCsvMutation } = useMutation(
-            [],
-            () => getGsheetLink(),
-            {
-                onMutate: () => {
-                    showToast({
-                        title: 'Exporting Google Sheets',
-                        subtitle: 'This may take a few minutes...',
-                        icon: (
-                            <Spinner
-                                className="bp4-icon bp4-icon-error"
-                                size={16}
-                            />
-                        ),
-                        key: 'exporting-gsheets',
-                        timeout: 0,
-                    });
-                },
-                onSuccess: (scheduledCsvResponse) => {
-                    pollCsvFileUrl(scheduledCsvResponse)
-                        .then((url) => {
-                            if (url) window.open(url, '_blank'); //Open in new tab
-                            AppToaster.dismiss('exporting-gsheets');
-                        })
-                        .catch((error) => {
-                            AppToaster.dismiss('exporting-gsheets');
-
-                            showToastError({
-                                title: `Unable to upload Google Sheets`,
-                                subtitle: error?.error?.message,
-                            });
-                        });
-                },
-                onError: (error: { error: Error }) => {
-                    AppToaster.dismiss('exporting-gsheets');
+        // TODO: move to own hook
 
-                    showToastError({
-                        title: `Unable to upload to Google Sheets`,
-                        subtitle: error?.error?.message,
-                    });
-                },
+        const {
+            mutateAsync: startGoogleSheetExport,
+            data: startGoogleSheetExportData,
+        } = useMutation(['google-sheets-start'], getGsheetLink, {
+            onMutate: () => {
+                showToast({
+                    title: 'Exporting Google Sheets',
+                    subtitle: 'This may take a few minutes...',
+                    icon: (
+                        <Spinner
+                            className="bp4-icon bp4-icon-error"
+                            size={16}
+                        />
+                    ),
+                    key: 'exporting-gsheets',
+                    timeout: 0,
+                });
             },
-        );
+            onError: (error: { error: Error }) => {
+                AppToaster.dismiss('exporting-gsheets');
+                showToastError({
+                    title: `Unable to upload to Google Sheets`,
+                    subtitle: error?.error?.message,
+                });
+            },
+        });
+
+        const { data: exportGoogleSheetData } = useQuery<
+            ApiDownloadCsv | undefined,
```

Comment:
2/2 THEN: this one just polls for data in the right location until it sees it?

Is there a failure on timeout set up somewhere that I don't see?

Reply:
Added a retry condition!